react/REACT1/week3/todo-list-app/src/todoitem.js

```diff
@@ -0,0 +1,79 @@
+import React from "react";
+import { useState } from "react";
+import BorderComponent from "./border";
+
+export default function TodoItem(prop) {
+  const doneFirstStatus = false;
+  const [done, setDone] = useState(doneFirstStatus);
+  const [update, setUpdate] = useState(false);
+  let [descriptionEdit, setDescriptionEdit] = useState(prop.description);
+
+  const editeInputHandler = (e) => {
+    setDescriptionEdit(e.target.value);
+    console.log(descriptionEdit);
+  };
+
+  const editTodo = () => {
+    setUpdate(true);
+  };
+
+  const updateTodo = () => {
```

Comment:
Would suggest avoiding putting complicated logic in your setState functions. Maybe instead it would be possible to take the todos as a prop, call the map on them and then use setState without a callback like in the comment before?